src/Model/Cli/BaseCliCommand.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace Waffle\Model\Cli;
+
+use Symfony\Component\Process\Process;
+use Exception;
+use Waffle\Traits\ConfigTrait;
+use Waffle\Model\Config\ProjectConfig;
+
+class BaseCliCommand
+{
+    use ConfigTrait;
+    
+    /**
+     * @var string[]
+     */
+    private $args = [];
+    
+    /**
+     * @var Process
+     */
+    private $process;
+    
+    /**
+     * A reference to the project config.
+     *
+     * @var ProjectConfig
+     */
+    protected $config;
+    
+    /**
+     * Constructor
+     *
+     * @param string[] The Arguments.
+     *
+     * @throws Exception
+     */
+    public function __construct(array $args)
+    {
+        if (empty($args)) {
+            throw new Exception('Invalid Arguments: You must pass at least one argument.');
+        }
+        
+        $this->config = $this->getConfig();
+        
+        // @todo: check for config prefix
+        if (!empty($this->config->getCommandPrefix())) {
```

Comment:
I like where this is going, but this is going to lock us into using a global prefix for all cli commands. That makes sense for using something like lando, but would prevent us from adding environment variables or anything else that may be specific to a particular tool we are using.

I have not thought through all of the scenarios that we may need to cover, so this is fine for now. I suspect this will need an update soon though.

Reply:
yeah, this was specifically for `lando` support, but I don't think it even works since I don't believe you need to put `lando` in front of Composer commands. It needs more testing/tweaking.